inc/zoo/swar/SWAR.h

```diff
@@ -8,23 +8,28 @@
 
 namespace zoo { namespace swar {
 
+using u64 = uint64_t;
+using u32 = uint32_t;
+using u16 = uint16_t;
+using u8 = uint8_t;
+
 /// Repeats the given pattern in the whole of the argument
 /// \tparam T the desired integral type
 /// \tparam Progression how big the pattern is
 /// \tparam Remaining how many more times to copy the pattern
 template<typename T, int Progression, int Remaining> struct BitmaskMaker {
-    constexpr static T repeat(T v) {
+    constexpr static T repeat(T v) noexcept {
```

Comment:
Thanks for making this "`noexcept`-correct"  John Lakos would say this is ridiculous but out of experience I know it is not, the noexceptness will eventually allow us to have guaranteed-success operations and those that are not.  One example: invoking "count leading zeroes" (binary logarithm) or trailing zeroes (index within the SWAR) are not guaranteed to work if the number is 0, this makes them not-safe.  We can make an algebraic system in which these things are automatically selected, for much better correctness and perhaps performance.

Reply:
noexcept as a concept is very powerful. I agree it is nice to set us up for a regime in which we can set up always most-correct behavior.